tests/ignite/metrics/test_psnr.py

```diff
@@ -123,100 +125,105 @@ def update(engine, i):
-    _test(y_pred, y, data_range, "cpu", n_iters, s, offset, rank, atol=atol)
+def _test_distrib_multilabel_input_YCbCr(device, atol=1e-8):
+    def get_test_cases():
 
-    # test with NHW shape
-    manual_seed(42)
-    y_pred = torch.rand(offset * idist.get_world_size(), 28, 28, device=device)
-    y = y_pred * 0.8
-    data_range = (y.max() - y.min()).cpu().item()
-    _test(y_pred, y, data_range, "cpu", n_iters, s, offset, rank, atol=atol)
+        y_pred = torch.randint(16, 236, (n_iters * batch_size, 1, 12, 12), dtype=torch.uint8, device=device)
+        cbcr_pred = torch.randint(16, 241, (n_iters * batch_size, 2, 12, 12), dtype=torch.uint8, device=device)
+        y = torch.randint(16, 236, (n_iters * batch_size, 1, 12, 12), dtype=torch.uint8, device=device)
+        cbcr = torch.randint(16, 241, (n_iters * batch_size, 2, 12, 12), dtype=torch.uint8, device=device)
 
-    if torch.device(device).type != "xla":
-        manual_seed(42)
-        y_pred = torch.rand(offset * idist.get_world_size(), 3, 28, 28, device=device)
-        y = y_pred * 0.65
-        data_range = (y.max() - y.min()).cpu().item()
-        _test(y_pred, y, data_range, idist.device(), n_iters, s, offset, rank, atol=atol)
-
-        # test for YCbCr
-        manual_seed(42)
-        y_pred = torch.randint(16, 236, (offset * idist.get_world_size(), 1, 12, 12), dtype=torch.uint8, device=device)
-        cbcr_pred = torch.randint(
-            16, 241, (offset * idist.get_world_size(), 2, 12, 12), dtype=torch.uint8, device=device
-        )
-        y = torch.randint(16, 236, (offset * idist.get_world_size(), 1, 12, 12), dtype=torch.uint8, device=device)
-        cbcr = torch.randint(16, 241, (offset * idist.get_world_size(), 2, 12, 12), dtype=torch.uint8, device=device)
         y_pred, y = torch.cat((y_pred, cbcr_pred), dim=1), torch.cat((y, cbcr), dim=1)
-        data_range = (y[:, 0, ...].max() - y[:, 0, ...].min()).cpu().item()
-        _test(
-            y_pred=y_pred,
-            y=y,
-            data_range=data_range,
-            metric_device=idist.device(),
-            n_iters=n_iters,
-            s=s,
-            offset=offset,
-            rank=rank,
-            atol=atol,
-            output_transform=lambda x: (x[0][:, 0, ...], x[1][:, 0, ...]),
-            compute_y_channel=True,
-        )
 
-        manual_seed(42)
-        y_pred = torch.randint(0, 256, (offset * idist.get_world_size(), 3, 16, 16), device=device, dtype=torch.uint8)
+        return y_pred, y
+
+    n_iters = 100
+    batch_size = 10
+
+    def out_fn(x):
+        return x[0][:, 0, ...], x[1][:, 0, ...]
+
+    rank = idist.get_rank()
+    for i in range(3):
+        # check multiple random inputs as random exact occurencies are rare
+        torch.manual_seed(42 + rank + i)
+        y_pred, y = get_test_cases()
+        _test(y_pred, y, 220, "cpu", n_iters, batch_size, atol, output_transform=out_fn, compute_y_channel=True)
+        if device.type != "xla":
+            _test(
```

Comment:
This line can not be written with 1 line, is there any method to deal with?

Reply:
Check this comment https://github.com/pytorch/ignite/pull/2679#discussion_r961583454 
It does not work ?